tripy/tripy/utils/ast.py

```diff
@@ -123,9 +123,14 @@ def get_arg_candidate_column_offsets(
             if is_kwarg:
                 arg_node = node.keywords[index - num_positional]
             else:
+                # For methods, the `self` argument is omitted from ast.Call.args
                 if "self" in arg_names:
-                    # For methods, the `self` argument is omited from ast.Call.args
-                    arg_node = node.args[index - 1]
+                    index -= 1
+                # If the final argument is a starred object, then we treat any args
+                # past the end as pointing to the starred object (this would be a variadic call,
+                # and the starred object would be a catchall)
+                if index >= len(node.args) and isinstance(node.args[-1], ast.Starred):
+                    arg_node = node.args[-1]
```

Comment:
This is a case that simply didn't come up before. Here, it results from `slice_helper(input_tensor, *args)` in the slice cases. I am not exactly sure why it did not come up with the previous approach, since the conditions should have been met there too. Perhaps the source code is being fetched when it should not have?

Reply:
Ah, this has to do with `STACK_DEPTH_OF_BUILD` in the tensor constructor. We may have to adjust it because the maximum number of stack frames is, I think, more than 4 now. For one, the tensor constructor now includes a call to `raw_init`, so it should be at least 5. Are there any other stack frames we should consider? I don't think these changes should add any since they just replace existing uses of the conversion decorator.

Reply:
Could you add a test for `STACK_DEPTH_OF_BUILD`? Its purpose is to exclude frames for helpers like `build_internal`, `raw_init`, etc. such that the first frame we see for `Tensor`s is the API (e.g. `full`) where `Op.build` was called.

Reply:
Would also be good to add a unit test for `get_arg_candidate_column_offsets`. I think all the logic in this file is very fragile and not well tested.

Reply:
Perhaps our behavior was wrong before. 4 does appear to be the correct number from the cases I've looked at, so we probably should have run into that case earlier.